### PR TITLE
Remove old users

### LINKED ISSUE #1021: Auto disable moloch user if not used
Now that we have users lastUsed time, add an optional ability to auto disable accounts when they haven't used moloch for X days.

* New config.ini setting that specifies X days of nonuse to auto disable
* When setting is set, in the cron enabled viewer, periodic check all enabled non admin users with a lastUsed date set, and set any that haven't used for X days as disabled
* Only check once a day
* A person that gets reenabled will have to use it right away, or they will get disabled the next run

### PR DESCRIPTION
**Pull request by 31453**
fixes #1021

### CODE REVIEW

**Comment by 31453** [comment]
Halp! Unknown territory! 🆘

File: db/db.pl [hunk 1]
```diff
@@ -2083,6 +2085,35 @@ sub parseArgs {
     }
     esPost("/_flush/synced", "", 1);
     logmsg "\n";
+    exit 0;
+} elsif ($ARGV[1] eq "rmusers") {
+    showHelp("Invalid number of <hours>") if (!defined $ARGV[2] || $ARGV[2] !~ /^[+-]?\d+$/);
```
Comment (awick) [inline db/db.pl]: should say days not hours

File: db/db.pl [hunk 2]
```diff
@@ -2083,6 +2085,35 @@ sub parseArgs {
...
+
+    my $users = esGet("/${PREFIX}users/_search?size=1000");
+    my $rmcount = 0;
+
+    foreach my $hit (@{$users->{hits}->{hits}}) {
+        if (exists $hit->{_source}->{lastUsed} && !$hit->{_source}->{createEnabled} && $hit->{_source}->{enabled}) {
```
Comment (awick) [inline db/db.pl]: this works, but you could also just have es do this with something like (not sure totally right), but either way is fine.

/${PREFIX}users/_search?size=1000&q=_exists_:lastUsed+AND+createEnabled:1+AND+enabled:1

https://www.elastic.co/guide/en/elasticsearch/reference/current/query-dsl-query-string-query.html

File: db/db.pl (hunk 1 above)
Comment (31453) [reply db/db.pl]: but duh!

File: db/db.pl (hunk 2 above)
Comment (31453) [reply db/db.pl]: 👍

**Pull request merged**